### ISSUE TITLE
Deprecate iOSVersionStyle.iOS13

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#62521]
`iOSVersionStyle.iOS13` was introduced in https://github.com/flutter/flutter/pull/60179.   Deprecate it so it's not available in the next stable release.

**Comment by jmagman** [comment]
\cc @justinmc @edwardaux @ctrysbita

**Comment by jmagman** [comment]
I'm kind of surprised this enum was introduced at all, I thought we had discussed holding off on all iOS 14 changes until the 1.20 stable branch, then updating to iOS 14 without options to fallback, the way it was done in previous years.
\cc @xster

**Comment by justinmc** [comment]
I think that should continue to be the plan for new OS features.  In this case I believe @ctrysbita specifically needed the enum, and planned to remove it by the time iOS 14 was released.

**Comment by jmagman** [comment]
If the deprecation lands in master now, it should get to stable coinciding with when iOS 14 releases.

**Comment by ctrysbita** [comment]
Hi, I think there are still some problems with our rules for new iOS features.

According to the statistics from [App Store Support](https://developer.apple.com/support/app-store/), 81% of all iPhone devices uses iOS13. There are still a lot of users keep using iOS12- or their devices cannot upgrade to iOS13. If the latest framework only support the behavior for latest version of iOS, our app may work quite different than native behavior on these devices (about 1/5) without any choice. Developers cannot make the app behave according to specific version iOS.

Our material widget library also keeps the deprecated material 2014 style and the old transition animation before Android 9 which means developers can switch the standard to use according to the actual OS version.

**Comment by justinmc** [comment]
I agree it would be nice to support old OS versions, but the reason why we have avoided it (besides a few cases) is because it's a big maintenance burden for Flutter.  It's tough just to keep up with the latest OS versions, and adding the need to maintain older versions as well is, we think for now, not the best use of limited resources.

In this case, I think it's fair to deprecate iOSVersionStyle now and default to iOS14 style.  We can wait to fully remove iOSVersionStyle until iOS14 adoption is the obvious majority.

**Comment by xster** [comment]
@ctrysbita our current stable has only the present styling and we expect the next stable release to roll out after iOS 14 is past golden master. We believe it shouldn't be a mainstream use case to have to toggle between the 2 styles actively on the same SDK revision. Do you have a specific use case for it?

**Comment by ctrysbita** [comment]
For an example.
Currently in my flutter app, I use `FadeUpwardsPageTransitions` for Android 8.0-, `OpenUpwardsPageTransitions` for Android 9.0+ and `CupertinoPageTransitions` for iOS. For each platform and OS version the app will behave like their native apps. 

But @justinmc is right, if our resources are quite limited, deprecate the old style and only support for the latest one would be better.

**Comment by jmagman** [comment]
@justinmc Can this be closed as of #62741?  
Let's never do this again...

**Comment by justinmc** [comment]
I think we should totally remove `iOSVersionStyle` altogether, and close this issue once that is done.  There was some discussion about this [here](https://github.com/flutter/flutter/pull/60179#discussion_r451689446).

@ctrysbita Are you available to open another PR removing `iOSVersionStyle` and the enum?  iOS14 is being released tomorrow, so it's probably time to start doing that.

**Comment by ctrysbita** [comment]
@justinmc Sure, I'm working for it.

**Comment by justinmc** [comment]
@ctrysbita Thanks for the quick response!  Tag me for review whenever you want.

**Comment by ctrysbita** [comment]
@justinmc Now this can be closed.

**Comment by justinmc** [comment]
Awesome, thank you!

**Comment by lukepighetti** [comment]
I don't believe this has been given the attention it deserves. At the very least we should commit to supporting the last two versions.
